netty/src/main/java/io/grpc/netty/ProtocolNegotiator.java

```diff
@@ -51,16 +51,16 @@
      * The HTTP/2 scheme to be used when sending {@code HEADERS}.
      */
     AsciiString scheme();
+
+    /**
+     * Returns a set of attributes, which varies depending on the handler implementation.
+     */
+    Attributes getAttributes();
```

Comment:
Why add a new method instead of using the existing `handleProtocolNegotiationCompleted`? As it is, this method is being called too early in `NettyClientTransport`; it is being called before the TCP connection has even been created. The only signal that the negotiation is complete is via the `handleProtocolNegotiationCompleted` callback.